src/sql/src/plan/statement/ddl.rs

```diff
@@ -1530,21 +1598,114 @@ pub fn plan_create_table_from_source(
+                sql_bail!("INCLUDE <metadata> requires ENVELOPE (NONE|UPSERT|DEBEZIUM)");
+            }
+
+            let metadata_columns = include_metadata
+                .into_iter()
+                .flat_map(|item| match item {
+                    SourceIncludeMetadata::Timestamp { alias } => {
+                        let name = match alias {
+                            Some(name) => name.to_string(),
+                            None => "timestamp".to_owned(),
+                        };
+                        Some((name, KafkaMetadataKind::Timestamp))
+                    }
+                    SourceIncludeMetadata::Partition { alias } => {
+                        let name = match alias {
+                            Some(name) => name.to_string(),
+                            None => "partition".to_owned(),
+                        };
+                        Some((name, KafkaMetadataKind::Partition))
+                    }
+                    SourceIncludeMetadata::Offset { alias } => {
+                        let name = match alias {
+                            Some(name) => name.to_string(),
+                            None => "offset".to_owned(),
+                        };
+                        Some((name, KafkaMetadataKind::Offset))
+                    }
+                    SourceIncludeMetadata::Headers { alias } => {
+                        let name = match alias {
+                            Some(name) => name.to_string(),
+                            None => "headers".to_owned(),
+                        };
+                        Some((name, KafkaMetadataKind::Headers))
+                    }
+                    SourceIncludeMetadata::Header {
+                        alias,
+                        key,
+                        use_bytes,
+                    } => Some((
+                        alias.to_string(),
+                        KafkaMetadataKind::Header {
+                            key: key.clone(),
+                            use_bytes: *use_bytes,
+                        },
+                    )),
+                    SourceIncludeMetadata::Key { .. } => {
+                        // handled below
+                        None
+                    }
+                })
+                .collect();
+
+            SourceExportDetails::Kafka(KafkaSourceExportDetails { metadata_columns })
+        }
+    };
+
+    let source_connection = &source_item.source_desc()?.expect("is source").connection;
+
+    // Some source-types (e.g. postgres, mysql, multi-output load-gen sources) define a value_schema
+    // during purification and populate the `columns` and `constraints` fields for the statement,
+    // whereas other source-types (e.g. kafka, single-output load-gen sources) do not, so instead
+    // we use the source connection's default schema.
+    let (key_desc, value_desc) = if !columns.is_empty() || !constraints.is_empty() {
```

Comment:
Ah good point on the pg source with empty columns. I'll make a fix for that.

The goal of this conditional was to accommodate both styles of how we've implemented purification/planning for our various source types to determine the output relations to pass to the encoding/envelope stage, without completely undertaking a refactor of where all these things happened (between purification and planning).

The first style is what we do for mysql/postgres and multi-output load-gen sources, which uses purification to retrieve some external schema and turn that into the appropriate column-types during purification. In those cases the columns/constraints are also stored on the purified statement in the catalog, such that here we just need to turn them into a RelationDesc.

The second style is what I found we did for kafka and single-output load-gen sources, which is not to do much work in purification at all, besides retrieving any schema-registry schemas and storing those in the purified statement. You'll see that the purification work we do for kafka 'tables' is really minimal, all we do is check that the external-reference is either not specified or is the topic and insert it if needed. This is why the kafka source exports can just use the `default` key / value desc, since those are standard across all 'outputs' and will be turned into the appropriate RelationDesc in the `apply_source_envelope_encoding` method below

Reply:
actually this should work fine for 'empty' postgres tables since the `default_value_desc` for postgres is an empty `RelationDesc`